src/libsystemd-network/network-internal.c

```diff
@@ -425,7 +425,7 @@ int deserialize_in_addrs(struct in_addr **ret, const char *string) {
         _cleanup_free_ struct in_addr *addresses = NULL;
         int size = 0;
 
-        assert(ret);
+        assert(ret && !*ret);
```

Comment:
We usually do not check like this, as if memory is leaked, then it is a caller issue.